libraries/AP_HAL_ChibiOS/hwdef/BrahmaF4/hwdef.dat

```diff
@@ -0,0 +1,141 @@
+OSCILLATOR_HZ 8000000
+
+STM32_ST_USE_TIMER 5
+
+# flash size
+FLASH_SIZE_KB 1024
+FLASH_RESERVE_START_KB 48
+define HAL_STORAGE_SIZE 16384
+define STORAGE_FLASH_PAGE 1
+
+################# UART Configuration ##################
+SERIAL_ORDER OTG1 USART1 USART3 USART6 UART4 
+
+PA11 OTG_FS_DM OTG1
+PA12 OTG_FS_DP OTG1
+
+PA10 USART1_RX USART1
+PA9 USART1_TX USART1
+define DEFAULT_SERIAL1_PROTOCOL SerialProtocol_RCIN
+
+PB10 USART3_TX USART3 NODMA
+PB11 USART3_RX USART3 NODMA
+
+PA0 UART4_TX UART4 NODMA
+PA1 UART4_RX UART4 NODMA
+
+PC6 USART6_TX USART6
+PC7 USART6_RX USART6
+
+##################SPI Configuration##############
+PA5 SPI1_SCK SPI1
+PA6 SPI1_MISO SPI1
+PA7 SPI1_MOSI SPI1
+PA4 OSD1_CS CS
+
+PB15 SPI2_MOSI SPI2
+PB14 SPI2_MISO SPI2
+PB13 SPI2_SCK SPI2
+PB12 FLASH1_CS CS
+
+PC12 SPI3_MOSI SPI3
+PC11 SPI3_MISO SPI3
+PC10 SPI3_SCK SPI3
+PB3 GYRO1_CS CS
+
+#SPI_DEVICES
+SPIDEV osd        SPI1 DEVID4  OSD1_CS  MODE0 10*MHZ 10*MHZ
+SPIDEV dataflash  SPI2 DEVID2 FLASH1_CS  MODE3 32*MHZ 32*MHZ
+SPIDEV imu1       SPI3 DEVID1  GYRO1_CS  MODE3  1*MHZ  4*MHZ
+DMA_NOSHARE SPI3*
+DMA_PRIORITY TIM1* TIM2* TIM2* TIM4* SPI3* TIM8*
+IMU BMI270 SPI:imu1 ROTATION_ROLL_180
+
+####################I2C bus Configuration##########
+#define HAL_USE_I2C TRUE
+#define STM32_I2C_USE_I2C1 TRUE
+I2C_ORDER I2C1
+PB6 I2C1_SCL I2C1 PULLUP
+PB7 I2C1_SDA I2C1 PULLUP
+BARO DPS310 I2C:0:0x77
+#define AP_BARO_BACKEND_DEFAULT_ENABLED 0
+#define HAL_BARO_ALLOW_INIT_NO_BARO 1
+
+# no built-in compass
+# external compass types
+define ALLOW_ARM_NO_COMPASS
+define HAL_PROBE_EXTERNAL_I2C_COMPASSES
+define HAL_I2C_INTERNAL_MASK 0
+define HAL_COMPASS_AUTO_ROT_DEFAULT 2
+
+#############GPIO Motors###############
+#PB1 TIM3_CH4  TIM3 PWM(1) GPIO(50)
+#PB0 TIM3_CH3  TIM3 PWM(2) GPIO(51)
+
+PB1 TIM1_CH3N  TIM1 PWM(1) GPIO(50)
+PB0 TIM1_CH2N  TIM1 PWM(2) GPIO(51)
+PA3 TIM2_CH4  TIM2 PWM(3) GPIO(52)
+PA2 TIM2_CH3  TIM2 PWM(4) GPIO(53)
+
+PB9 TIM4_CH4  TIM4 PWM(5) GPIO(54) 
+PB4 TIM3_CH1  TIM3 PWM(6) GPIO(55)              
+PB5 TIM3_CH2  TIM3 PWM(7) GPIO(56)
+#PC9 TIM8_CH4  TIM8 PWM(8) GPIO(57)
+PC9 TIM3_CH4  TIM3 PWM(8) GPIO(57)
+
+PC8 TIM8_CH3  TIM8 PWM(9) GPIO(58) # LED NOPIXEL
+
+#PA15 TIM2_CH1 TIM2 RCININT PULLDOWN LOW
+
+# Motor order implies Betaflight/X for standard ESCs
+define HAL_FRAME_TYPE_DEFAULT 12
+
+#ADC definitions
+PC3 BATT_CURRENT_SENS ADC1 SCALE(1)
+define HAL_BATT_CURR_PIN 13
+define HAL_BATT_CURR_SCALE 37.0
+
+PC4 BATT_VOLTAGE_SENS ADC1 SCALE(1)
+define HAL_BATT_VOLT_PIN 14
+define HAL_BATT_VOLT_SCALE 11.0
+
+PC5 RSSI_ADC ADC1
+define BOARD_RSSI_ANA_PIN 15
+define HAL_BATT_MONITOR_DEFAULT 4
+
+
+#################Indicator LEDs#######################
+define AP_NOTIFY_GPIO_LED_2_ENABLED 1
+PA8 LED_YELLOW OUTPUT LOW GPIO(1)
+define HAL_GPIO_A_LED_PIN 1
+define HAL_GPIO_B_LED_PIN 2 
+
+
+# enable logging to dataflash
+define HAL_LOGGING_DATAFLASH_ENABLED 1
+
+######################osd setup######################
+define OSD_ENABLED 1
+define HAL_OSD_TYPE_DEFAULT 1
```

Comment:
you could setup as type 5 and enable both analog and DisplayPort...but as is you could add OSD_TYPE2 = 5 to default params